### PR TITLE
Ellblock memory

### PR DESCRIPTION
**Pull request by nicolasbock**
This change is [Reviewable](https://reviewable.io/reviews/lanl/bml/344)

### CODE REVIEW

File: src/C-interface/ellblock/bml_allocate_ellblock_typed.c [hunk 1]
```diff
@@ -183,13 +183,17 @@ bml_matrix_ellblock_t *TYPED_FUNC(
     int mb = bml_get_mb();
     int nb = bml_get_nb();
 
+    printf("N = %d, M = %d, len(bsize) = %d\n", N, M, nb);
```
Comment (jeanlucf22) [inline src/C-interface/ellblock/bml_allocate_ellblock_typed.c]: This type of debugging info should not be merged with master!

Comment (nicolasbock) [reply src/C-interface/ellblock/bml_allocate_ellblock_typed.c]: Definitely not @jeanlucf22 . My intention was to show you what debug statements I had inserted, and I should have marked this PR clearly as "do not merge".

**Comment by nicolasbock** [comment]
@jeanlucf22 I removed the debug print statements. Could you have a look at the changes? Thanks!

**Comment by jeanlucf22** [comment]
Looks fine to me. Note that I have in mind to create a memory pool for some of these allocations to be more efficient. I hope to get to work on it in the next few weeks.

**Comment by nicolasbock** [comment]
Nice! So you don't have to go through the system's `malloc`?

**Comment by jeanlucf22** [comment]
I would do one large N*M malloc, and then have a pointer to a piece of memory for each block

**Comment by nicolasbock** [comment]
I am interested in learning how that would perform.

**Pull request merged**